packages/appium/lib/appium.js

```diff
@@ -516,6 +524,17 @@ class AppiumDriver extends BaseDriver {
     // handling the command and which didn't
     plugins.length && this.logPluginHandlerReport({cmd, cmdHandledBy});
 
+    // And finally, if the command was createSession, we want to migrate any plugins which were
+    // previously sessionless to use the new sessionId, so that plugins can share state between
+    // their createSession method and other instance methods
+    if (cmd === 'createSession' && this.sessionlessPlugins.length && !res.error) {
+      const sessionId = res.value[0];
```

Comment:
since you used `_.last()`, what about `_.first()`? :wink:

Reply:
ok fine :-)